### DISCUSSION TITLE
debugging code linked to AMReX hangs

### DISCUSSION THREAD

**Discussion by tesari** [AMReX-Codes/amrex#3298] [General]
Hi,

I have AMReX build as a library in debug mode. When debugging my code, the debugger hangs at ‘New Thread …’:
	…
	Starting program: my_program
	[New Thread 0x5307 of process 87690]

I am aware that I need to change the values of throw_exception and signal_handling, but I initialize MPI with 

	amrex::Initialize(glb::myComm);

and cannot figure out how to do that. Could you advise, please?

Thanks,
Ted

**Comment by WeiqunZhang** [comment]
This might be a bug in gdb or macOS. https://apple.stackexchange.com/questions/420492/gdb-hangs-after-new-thread-on-macos. So probably no changes in amrex can help. If you want to disable signal handling and enable exception without using amrex::ParmParse, you can do
```
amrex::system::signal_handling = false;
amrex::system::error_handler = nullptr;
amrex::system::throw_exception = true;
```

**Comment by tesari** [comment]
Thanks a lot!
Ted

From: Weiqun Zhang ***@***.***>
Date: Friday, May 5, 2023 at 1:36 PM
To: AMReX-Codes/amrex ***@***.***>
Cc: Sariyski, Tzvetan Emilov ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [AMReX-Codes/amrex] debugging code linked to AMReX hangs (Discussion #3298)

This might be a bug in gdb or macOS. https://apple.stackexchange.com/questions/420492/gdb-hangs-after-new-thread-on-macos. So probably no changes in amrex can help. If you want to disable signal handling and enable exception without using amrex::ParmParse, you can do

amrex::system::signal_handling = false;

amrex::system::error_handler = nullptr;

**Reply by WeiqunZhang** [reply]
Note that I updated my comment.
